tests/IntervalTest.php

```diff
@@ -23,15 +23,26 @@ public function testEndInstantIsNotBeforeStartInstant(): void
         $this->expectException(DateTimeException::class);
         $this->expectExceptionMessage('The end instant must not be before the start instant.');
 
-        new Interval($end, $start);
+        Interval::of($end, $start);
     }
 
     public function testGetStartEnd(): void
     {
         $start = Instant::of(2000000000, 987654321);
         $end = Instant::of(2000000009, 123456789);
 
-        $interval = new Interval($start, $end);
```

Comment:
You should keep `new` here, otherwise the test is the same as `testGetStartEndUsingOf()`!

Reply:
Ah yes, haven't noticed it

Reply:
Since the constructor is deprecated, I think we can remove `testGetStartEndUsingOf()` test. I can add a single test which specifically tests that the public constructor still works though.

Reply:
> I can add a single test which specifically tests that the public constructor still works though.

That's this test :slightly_smiling_face: But you can rename it to make it more obvious.